pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -198,6 +202,27 @@ func (t *Tracer[Event]) Attach(pid uint32, eventCallback func(*Event)) error {
 	return nil
 }
 
+func (t *Tracer[Event]) SetEventHandler(handler any) {
+	nh, ok := handler.(func(ev *Event))
+	if !ok {
+		panic("event handler invalid")
+	}
+	t.eventHandler = nh
+}
+
+func (t *Tracer[Event]) AttachContainer(container *containercollection.Container) error {
+	return t.Attach(container.Pid, func(ev *Event) {
+		if setter, ok := any(ev).(operators.ContainerInfoSetters); ok {
+			setter.SetContainerInfo(container.Podname, container.Namespace, container.Name)
+		}
+		t.eventHandler(ev)
+	})
```

Comment:
Same issue here. I did some experiments in https://github.com/inspektor-gadget/inspektor-gadget/commit/b0fdc56bf04de70bc7be17dd1f076b7992ed4271 to follow the same logic we had before with [EnrichByNetNs](https://github.com/inspektor-gadget/inspektor-gadget/blob/b6c912061fb1bcf755921932d0a95074b8919348/pkg/container-collection/container-collection.go#L366-L388).

Reply:
In the following logic, we should verify if `EnrichEventByMntNs` has already enriched the event with all the container information before calling `EnrichEventByNetNs`. Otherwise, we risk removing the container name [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/b0fdc56bf04de70bc7be17dd1f076b7992ed4271/pkg/container-collection/operator.go#L64):

https://github.com/inspektor-gadget/inspektor-gadget/blob/b0fdc56bf04de70bc7be17dd1f076b7992ed4271/pkg/operators/localmanager/localmanager.go#L302-L313

To do so, we could make the [`CommonData`](https://github.com/inspektor-gadget/inspektor-gadget/blob/b0fdc56bf04de70bc7be17dd1f076b7992ed4271/pkg/types/types.go#L65) implement something like this:

```golang
func (c *CommonData) IsEnriched() bool {
	return c.Pod != "" && c.Namespace != "" && c.Container != ""
}
```

WDYT?

Reply:
Or avoid [`SetContainerInfo`](https://github.com/inspektor-gadget/inspektor-gadget/blob/b0fdc56bf04de70bc7be17dd1f076b7992ed4271/pkg/types/types.go#L85-L89) setting the values when the passed string is empty.